pkg/runtime/cache/namespace.go

```diff
@@ -72,40 +76,49 @@ func NewNamespaceCache(clientset kubernetes.Interface, log logr.Logger) *Namespa
 	return &NamespaceCache{
 		informer:       sharedInformer,
 		log:            log.WithName("cache.namespace"),
+		listenNamespace: listenNamespace,
 		namespaceInfos: make(map[string]*namespaceInfo),
 	}
 }
 
-// isIgnoredNamespace returns true if an object is of type corev1.Namespace
-// and it metadata name is one of 'ack-system', 'kube-system' or 'kube-public'
-func isIgnoredNamespace(raw interface{}) bool {
+// Check if the provided namespace should be listened to or not 
+func isListenNamespace(raw interface{}, listenNamespace string) bool {
 	object, ok := raw.(*corev1.Namespace)
-	return ok &&
-		(object.ObjectMeta.Name == "ack-system" ||
-			object.ObjectMeta.Name == "kube-system" ||
-			object.ObjectMeta.Name == "kube-public")
+	if !ok {
+		return false
+	} 
+
+	if listenNamespace == object.ObjectMeta.Name {
+		return true
+	} else if object.ObjectMeta.Name == "kube-system" || object.ObjectMeta.Name == "kube-public" {
+		return false
+	} else if listenNamespace == "" {
+		return true
+	}
```

Comment:
nit: we can simplify this if/else block to something like:
```go
	if listenNamespace != "" {
		return listenNamespace == object.ObjectMeta.Name
	}
	return object.ObjectMeta.Name != "kube-system" && object.ObjectMeta.Name != "kube-public"
```

Reply:
done